### DISCUSSION TITLE
Security concern regarding deleted messages

### DISCUSSION THREAD

**Discussion by dranada** [victornpb/undiscord#305] [1. Questions & Answers]
Hi there,

just want to confirm that the messages deleted with this script are not stored anywhere? Once I press clear on the logs, is all the information gone?

Thank you

**Comment by Toby222** [comment]
The userscript does not store anything.
It's equivalent to manually deleting your messages, so the only place you'd find message leftovers would be Discord itself.

**Reply by dranada** [reply]
Thank you very much for the response Toby!! So after I have ran the script and I clear the logs, the messages cease to exist? Are the logs themselves stored? And I assume our info isn't forwarded to third parties? Really appreciate your response! 

Also, sorry, I was expecting a response from Victor, but did you help with the script as well? I was wondering if this was confirmed by Victor himself

**Reply by Toby222** [reply]
Logs are only stored ephemerally in the browser console and in the output window of the plugin itself.

And the nice thing with open source (especially userscripts) is that anyone can confirm, if they understand the source code :p
cf. [here](https://github.com/victornpb/deleteDiscordMessages/blob/80b97efbed7091fe40138ec3f49b5113e1355bd5/deleteDiscordMessages.user.js#L53) and [here](https://github.com/victornpb/deleteDiscordMessages/blob/80b97efbed7091fe40138ec3f49b5113e1355bd5/deleteDiscordMessages.user.js#L448) to see how logging is done. (It literally writes to the console and the `pre` element that you see in the browser)
cf. [here](https://github.com/victornpb/deleteDiscordMessages/blob/80b97efbed7091fe40138ec3f49b5113e1355bd5/deleteDiscordMessages.user.js#L170) to see the API call that actually deletes the message. You can look around to see that the message nor any part of it is sent to anywhere.

**Reply by Toby222** [reply]
TL;DR: Nothing this script does is logged anywhere except by Discord Inc. itself

**Reply by dranada** [reply]
Thank you, I really appreciate your response - I dont have too much of a tech background so I cannot vet the script myself. So for privacy purposes, this script is completely safe to use? Once the logs are cleared, there are absolutely no remnants of the msgs?

**Reply by dranada** [reply]
Wait sorry, what does discord log on their end?

**Reply by Toby222** [reply]
Once you clear the log there's gonna be just as much remains as if you had manually went through and deleted those messages.

**Reply by Toby222** [reply]
> Wait sorry, what does discord log on their end?

Dates when users interacted with something, calls to their API, stuff like that.
The messages themselves will probably remain (but inaccessible) for up to a few minutes after deletion because it needs to be synced between their various servers.

From a very paranoid privacy perspective, it might be possible for someone with full access to Discord's backend to look at your API usage and determine that you were deleting a bunch of messages at the time you did.
The message *contents* are ultimately deleted. (If not, Discord Inc. would be in very hot legal trouble)

**Reply by dranada** [reply]
Oh that stuff is fine! Thank you for clarifying! As long as this script itself isn't storing any deleted data, it should be fine I think
